### DISCUSSION TITLE
Updating GitHub Classroom roster

### DISCUSSION THREAD

**Discussion by pinventado** [community/community#30850] [GitHub Education]
If you've previously uploaded a roster and want to add new students, make sure you only include the new students. I thought updating meant keeping old students, but apparently it creates a new entry if the name exists. For example, if a previously added student id is "Stu Dent," and your updated list contains the same person, the new roster would contain "Stu Dent" and "Stu Dent-1". I had 150 students so now I have 300 and am manually deleting duplicates T_T

Hopefully this saves someone the time from deletion like I did. 

If I might make a suggestion to the GitHub Education team ... it would be good to update the instructions to say that updating students means it will not delete existing students and would create duplicates if they exist.

**Comment by dham** [comment]
I got burnt by this this year too. I'd go so far as to say that the current behaviour is a bug. The class roster is made up of institution identifiers. It's reasonable to assume that these are unique, which means that if the same identifier is added twice this does refer to the same student, and does not mean that the instructor intended for that student to appear twice.

The current setup means that if students join a class the instructor has to somehow (probably manually) create a list of new IDs to add to the roster. The far more natural workflow would be to simply export the full list of identifiers from the institutional IT system and update the roster.

**Reply by iamabhutta** [reply]
This issue still exists. I would think they would learn to do a basic roster management by now!

**Reply by Teddyzander** [reply]
Is there any update on a workflow for this? I run a class with >800 students (!) and manually unlinking and then re-linking is such an insane chore for a class of this magnitude. I want to be able to download the classroom roster, make local edits, and then re-upload the roster (or similar) so that I can bulk edit the ID<-->github_username link.

I also do not think students should be allowed to "skip" linking their account (unless the admin specically allows it!)
